---
title: "Water Lilies Floating and Buried 16S"
author: "Brianne Palmer"
date: "2023-02-01"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir ="/Users/briannepalmer/Library/CloudStorage/OneDrive-Personal/LeafFossilization/Leaf Fossilization R/Experiment 7")
```
There are several packages required to perform these analyses and create the figures. There are probably easier ways to do some of this, but if the code works, why change it?
```{r, Load Libraries}
library(phyloseq)
library(plyr)
library(data.table)
library(tidyverse)
library(vegan)
library(ecodist)
library(patchwork)
library(RColorBrewer)
library(ape)
library(microbiome)
library(microbiomeMarker)
library(ampvis2)
library(ggpubr)
library(ggvenn)
library(indicspecies)
library(ggpattern)
library(rstatix)
library(ggrepel)
```

Load Data
```{r}
S16 <- fread("/Users/briannepalmer/Library/CloudStorage/OneDrive-Personal/LeafFossilization/Leaf Fossilization R/Experiment 7/data/Experiment7/waterlilies_w_soilwater_03april.csv")

#S16 <- S16 %>% filter(genus != "", genus != "g__uncultured")
```

```{r}
# the metadata 
metadata <- fread("/Users/briannepalmer/Library/CloudStorage/OneDrive-Personal/LeafFossilization/Leaf Fossilization R/Experiment 7/data/Experiment7/experiment7_metadata.csv")
```

```{r}
#ASV.16 <- as.vector(paste0("ASV", x16))
#S16$ASV <- ASV.16
```

```{r}
#S16.wide[,c(9:26)] <-sweep(S16.wide[,c(9:26)], 2, colSums(S16.wide[,c(9:26)]), `/`)
#S16.wide$n <- rowSums(S16.wide[ , c(9:26)], na.rm=TRUE)

S16[,c(9:26)][is.na(S16[,c(9:26)])] <- 0
```

Make another table with the ASVs as columns, this is necessary for the PCOA

```{r, PCOA for 16S}
library(funrar)

transformed.16 <- t(S16[,c(9:26)])
#colnames(transformed.16) <- ASV.16
transformed.16[is.na(transformed.16)] <-0
transformed.16[c(1:185724)] <- sapply(transformed.16[c(1:185724)],as.numeric)

S16.rel <- make_relative(transformed.16)

otu.bray.16 <- vegdist(S16.rel, method = "bray")
pcoaVS.16 <- ecodist::pco(otu.bray.16, negvals = "zero", dround = 0)

pcoa1.16 = pcoaVS.16$vectors[,1] # adds the points of the NMDS 1 dimmension
pcoa2.16 = pcoaVS.16$vectors[,2] # adds the points of the NMDS 2 dimmension

MDS = data.frame(pcoa1.16 = pcoa1.16, pcoa2.16 = pcoa2.16)
MDS = cbind(MDS, metadata)

set.seed(12345)
adonis2(transformed.16 ~ Treatment, metadata) 
adonis2(transformed.16 ~ Type, metadata) 
adonis2(transformed.16 ~ Location, metadata)
adonis2(transformed.16 ~ Plant, metadata)
```
```{r}
pcoa <- ggplot(data = MDS, aes(x = pcoa1.16, y = pcoa2.16, shape = Treatment)) + geom_point(size =3, color = '#EE8866') + theme_bw() + theme(text=element_text(size=20)) +labs(x = "PCOA1", y = "PCOA2") + xlim(-0.3, 0.3)

pdf("figures/Experiment7/NymphaeaNuphar_Figures_April2023/pcoa_16S.pdf")
pcoa
dev.off()
```

```{r, PCOA for 16S}
transformed.16.leaves <- transformed.16[-c(1:6),]

S16.rel.leaves <- make_relative(transformed.16.leaves)

otu.bray.16 <- vegdist(S16.rel.leaves, method = "bray")
pcoaVS.16 <- ecodist::pco(otu.bray.16, negvals = "zero", dround = 0)

pcoa1.16 = pcoaVS.16$vectors[,1] # adds the points of the NMDS 1 dimmension
pcoa2.16 = pcoaVS.16$vectors[,2] # adds the points of the NMDS 2 dimmension

metadata.leaves <- metadata %>% filter(Plant != "None")
MDS = data.frame(pcoa1.16 = pcoa1.16, pcoa2.16 = pcoa2.16)
MDS = cbind(MDS, metadata.leaves)

set.seed(12345)
adonis2(transformed.16.leaves ~ Treatment, metadata.leaves) 
adonis2(transformed.16.leaves ~ Location, metadata.leaves)
adonis2(transformed.16.leaves ~ Plant, metadata.leaves)
```

Time to make some plots, but first, I want to make a color scheme for all the future figures. Hopefully, it is color blind friendly 
```{r}
pal <- palette(c('#882255','#88CCEE','#EE8866','#44AA99','#117733','#332288','#DDCC77','#999933','#CC6677','#AA4499'))
largepal <- colorRampPalette(pal)(50) # for large number of variables
two.pal <- palette(c('#882255','#88CCEE'))
```

```{r, echo=FALSE}
ggplot(data = MDS, aes(x = pcoa1.16, y = pcoa2.16, shape = Treatment)) + geom_point(size =3) + theme_bw() + theme(text=element_text(size=20)) +labs(x = "PCOA1", y = "PCOA2") 
```

Look at the abundance of each phylum and make bar charts for the phyla within each kingdom 

For this, we have to make another table, I am calling it pivot here but it is a transposed table essentially 
```{r, 16S}
# remove ASVs that have no information 
pivot.16S <- S16 %>% filter(kingdom != "Unassigned", kingdom != "Alveolata") %>% pivot_longer(cols = c(BS1:NyTopR3), names_to = "SampleID")

# if there are blanks, these should be NA
pivot.16S[pivot.16S == ""] <- NA  
pivot.16S$value[is.na(pivot.16S$value)] <- 0

pivot.16S <- pivot.16S %>% group_by(SampleID, kingdom, phylum, class, order, family, genus, species) %>% dplyr::summarize(value = sum(value))

phylum.16S <- pivot.16S %>% dplyr::select(SampleID, kingdom, phylum, value) %>% group_by(SampleID, kingdom, phylum) %>% dplyr::summarize(value = sum(value))

phylum.16S <- phylum.16S %>% pivot_wider(names_from = phylum, values_from = value)

phylum.16S[c(3:60)] <- sapply(phylum.16S[c(3:60)],as.numeric)
phylum.16S[is.na(phylum.16S)] <- 0

phylum.16S <- merge(metadata, phylum.16S, by = "SampleID")

# make a longer table, this is easier to graph with (hence why I named it ggplot)
phylum.ggplot.16S <- pivot_longer(phylum.16S, cols = p__Crenarchaeota:p__Zixibacteria,names_to = "phylum", values_to = "value") %>% mutate(rel = value/sum(value))

one.percent <- phylum.ggplot.16S %>% dplyr::group_by(phylum) %>% dplyr::summarise_at(.vars = "rel", sum) %>% filter(rel < 0.01)
one.percent.vector <- phylum.ggplot.16S %>% filter(phylum %in% one.percent$phylum)
`%notin%` <- Negate(`%in%`)
phylum.ggplot <- phylum.ggplot.16S %>% filter(phylum %notin% one.percent$phylum)
one.percent.vector$phylum <- rep("< 1% Relative Abundance", nrow(one.percent.vector))

phylum.ggplot.16S <- rbind(phylum.ggplot.16S, one.percent.vector)
#phylum.ggplot.16S <- merge(phylum.ggplot.16S, metadata, by = "SampleID")

phylum.group <- pivot.16S %>% group_by(phylum) %>% summarise(value = sum(value)) 
phylum.group <- phylum.group[order(-phylum.group$value),]
phylum.10 <- phylum.group[c(1:10),]
```

```{r}
phylum.10.plot <- ggplot(
    phylum.ggplot.16S %>% filter(phylum %in% phylum.10$phylum),
    aes(fill = phylum, y = value, x = Treatment)
  ) + geom_bar(position = "fill", stat = "identity")  + theme(axis.text.x = element_text(size = 10)) + theme(legend.position = "right") + labs(y = "Relative Abundance") + theme_bw() + ggtitle("Prokaryotic Phyla") + scale_fill_manual(values = pal) +
  theme(axis.text.x = element_text(
    angle = 45,
    vjust = 1,
    hjust = 1
  ))

pdf("figures/Experiment7/NymphaeaNuphar_Figures_April2023/phylum_barchart_16S.pdf", height = 12, width = 18)
phylum.10.plot
dev.off()
```


```{r}
phylum.anova <- phylum.ggplot.16S %>% group_by(phylum) %>%
  do(broom::tidy(TukeyHSD(aov(value ~ Treatment, data = .)))) %>%
  ungroup

phylum.anova.sig <- phylum.anova %>% filter(adj.p.value < 0.05) 

phylum.anova.sig # 
```

```{r}
S16.2 <- fread("/Users/briannepalmer/Library/CloudStorage/OneDrive-Personal/LeafFossilization/Leaf Fossilization R/Experiment 7/data/Experiment7/waterlilies_w_soilwater_03april.csv")
x16 <- seq(1:10318)

ASV.16 <- as.vector(paste0("ASV", x16))
S16.2$ASV <- ASV.16

# if there are blanks, these should be NA
S16.2[,c(9:26)][is.na(S16.2[,c(9:26)])] <- 0

#S162 <- S162 %>% filter(genus != "", genus != "g__uncultured")

otu.table <- S16.2[,c(9:26)]
otu.table <- as.matrix(otu.table)
rownames(otu.table) <- S16.2$ASV
tax.table <- S16.2[,c(2:8)]
tax.table <- as.matrix(tax.table)
rownames(tax.table) <- S16.2$ASV

otu.table <- otu_table(otu.table, taxa_are_rows = TRUE)
tax.table <- tax_table(tax.table)
sam.data <- as.matrix(metadata)
sam.data <- sample_data(metadata)
rownames(sam.data) <- metadata$SampleID

ps <- phyloseq(otu.table, tax.table, sam.data)
random_tree = rtree(ntaxa(ps), rooted=TRUE, tip.label=taxa_names(ps))

ps = merge_phyloseq(ps,random_tree)
ps <- subset_taxa(ps, kingdom !="Unassigned")
ps <- subset_taxa(ps, phylum != "NA")
#ps <- subset_taxa(ps, genus != "g__uncultured")
#ps <- subset_taxa(ps, genus != "NA")

p = plot_richness(ps, x="Treatment", color ="Treatment", measures=c("Observed","Shannon", "Simpson")) + geom_point() + theme_bw() + geom_boxplot() + scale_color_manual(values = pal) + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1), legend.position = "none") 

pdf("figures/Experiment7/NymphaeaNuphar_Figures_April2023/diversityplot_16S.pdf")
p
dev.off()
```
Make a new dataframe with the ASVs from the pivot dataframe
```{r}
leaves.only <- S16.2 %>% filter(kingdom != "Unassigned", kingdom != "Alveolata") %>% pivot_longer(cols = c(NuBotR1:NyTopR3), names_to = "SampleID")

leaves.only <- leaves.only %>% group_by(SampleID, kingdom, phylum, class, order, family, genus, species, ASV) %>% dplyr::summarize(value = sum(value))

#pivot.16S.2 <- pivot.16S.2 %>% filter(SampleID != "BS1", SampleID != "BS2",SampleID != "BS3",SampleID != "BW1",SampleID != "BW2",SampleID != "BW3" )

ASV <- leaves.only %>% dplyr::select(SampleID, ASV, value) # 
ASV <- ASV %>% pivot_wider(names_from = ASV, values_from = value)
ASV[c(9:10323)] <- sapply(ASV[c(9:10323)],as.numeric)
ASV[c(9:10323)][is.na(ASV[c(9:10323)])] <- 0
ASV <- merge(metadata, ASV)
ASV.gp <- ASV %>% group_by(SampleID, Location, Plant, Treatment) %>% summarize_at(vars(ASV2366:ASV915), sum)
```

```{r}
ASV.long <- pivot_longer(ASV.gp, cols = ASV2366:ASV915, names_to = "ASV", values_to = "count")

y <- ASV.long %>% dplyr::select(Treatment, ASV, count)

NuBot = y %>% filter(Treatment == "NupharBuried")  %>% filter(count >0) %>% dplyr::select(ASV) #3191
NuTop = y %>% filter(Treatment == "NupharFloating")  %>% filter(count >0) %>% dplyr::select(ASV) #1952
NyBot = y %>% filter(Treatment == "NymphaeaBuried")  %>% filter(count >0) %>% dplyr::select(ASV) #3111
NyTop = y %>% filter(Treatment == "NymphaeaFloating")  %>% filter(count >0) %>% dplyr::select(ASV) #1440

y <- rbind(NuBot, NuTop, NyBot, NyTop)

Treatment <- c(rep("NupharBuried", 3191), rep("NupharFloating", 1952),rep("NymphaeaBuried", 3111), rep("NymphaeaFloating", 1440))

y$Treatment <- Treatment

x <- list(NuBur = y$ASV[1:3191] ,NuFlo= y$ASV[3192:5143], NyBur= y$ASV[5144:8254] ,NyFlo = y$ASV[8255:9694])

```

```{r, echo=FALSE}
venn <- ggvenn(
  x, 
  fill_color = pal,
  stroke_size = 0.5, set_name_size = 3.5, show_percentage = TRUE) 

pdf("figures/Experiment7/NymphaeaNuphar_Figures_April2023/venn_16S.pdf")
venn
dev.off()
```


```{r}
#ps <- subset_taxa(ps, genus != "g__uncultured")
#ps <- subset_taxa(ps, genus != "NA")

otu.table <- as.data.frame(ps@otu_table)
tax.table <- as.data.frame(ps@tax_table)

d <- amp_load(otutable = otu.table,
              metadata = metadata,
              taxonomy = tax.table)

amp_rarecurve(
  d,
  stepsize = 1000,
  color_by = "SampleID",
  facet_by = NULL,
  facet_scales = "fixed"
)


d.rare <-amp_rarefy(d, 65597)

amp_heatmap(data = d.rare, 
             tax_aggregate = "Genus",
             tax_add = "Phylum",
             group_by = c("SampleID"),
             tax_show = 25,
             plot_colorscale = "log10")

alpha <- amp_alpha_diversity(d.rare)

rich.lm <- lm(ObservedOTUs ~ Plant, data = alpha %>% filter(Plant != "None"))
rich.aov <- aov(rich.lm)
TukeyHSD(rich.aov)
summary(rich.aov)

div.lm <- lm(Shannon ~ Location, data = alpha %>% filter(Plant != "None"))
div.aov <- aov(div.lm)
summary(div.aov)

div.lm <- lm(Simpson ~ Location, data = alpha%>% filter(Plant != "None"))
div.aov <- aov(div.lm)
summary(div.aov)


amp_heatmap(d.rare,
            group_by = c("Location"),
            tax_aggregate = "Genus",
            plot_functions = TRUE,
            tax_show = 25,
            functions = c("Filamentous", "Chemoautotroph_mixotroph", "AOB", "NOB", "Aerobic heterotroph", "Nitrite reduction", "Sulfate reduction", "Fermentation", "Acetogen", "Short-chain fatty acids", "Sugars", "Proteins_amino acids"))

tax_vector <- c("g__uncultured", "NA")
d.known.genera <- amp_filter_taxa(d.rare, 
                                  tax_vector = tax_vector, 
                                  remove = TRUE)

genus.heatmap <- amp_heatmap(data = d.known.genera, 
             tax_aggregate = "Genus",
             tax_add = "Phylum",
             group_by = c("Treatment"),
             tax_show = 25,
             plot_colorscale = "log10", 
             plot_values = FALSE)

pdf("figures/Experiment7/NymphaeaNuphar_Figures_April2023/heatmap_16S.pdf", width = 20, height = 20)
genus.heatmap
dev.off()

```

```{r}
amp_otu_network(
  d,
  min_abundance = 0,
  color_by = "treatment",
  tax_aggregate = "Phylum",
  tax_add = NULL,
  tax_show = 10,
  tax_class = NULL,
  tax_empty = "best",
  normalise = TRUE
)


amp_venn(
  d.rare,
  group_by = "Location",
  cut_a = 0.1,
  cut_f = 80,
  text_size = 5,
  normalise = TRUE,
  detailed_output = FALSE
)

```

Look at richness and diversity between all samples
```{r}
ggplot(data = alpha, aes(x = SampleID, y = ObservedOTUs, color = Location, shape = Plant)) + geom_point(size = 5) + theme_bw() + theme(legend.position = "none")  + labs(y = "# of ASV", x = "Location") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + theme_bw() + scale_color_manual(values = two.pal) + theme(text=element_text(size=20)) + theme (axis.text.x = element_text (angle = 45, vjust = 1, hjust=1)) 

ggplot(data = alpha, aes(x = Treatment, y = ObservedOTUs, fill = Location)) + geom_violin() + theme_bw() + theme(legend.position = "none")  + labs(y = "# of ASV", x = "Location") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + theme_bw() + scale_fill_manual(values = two.pal) + theme(text=element_text(size=20)) + theme (axis.text.x = element_text (angle = 45, vjust = 1, hjust=1)) 

ggplot(data = alpha, aes(x = SampleID, y = Shannon, color = Location, shape = Plant)) + geom_point(size = 5) + theme_bw() + theme(legend.position = "none")  + labs(y = "Shannon Diversity", x = "Location") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + theme_bw() + scale_color_manual(values = two.pal) + theme(text=element_text(size=20)) + theme (axis.text.x = element_text (angle = 45, vjust = 1, hjust=1)) 

ggplot(data = alpha, aes(x = Treatment, y = Shannon, fill = Location)) + geom_violin() + theme_bw() + theme(legend.position = "none")  + labs(y = "Shannon Diversity", x = "Location") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + theme_bw() + scale_fill_manual(values = two.pal) + theme(text=element_text(size=20)) + theme (axis.text.x = element_text (angle = 45, vjust = 1, hjust=1)) 
```

DESEQ
```{r}
ps1 <- subset_samples(ps, Plant != "None")
wh0 = genefilter_sample(ps1, filterfun_sample(function(x) x > 5), A=0.5*nsamples(ps1)) # removes low abundance pathways
GP1 = prune_taxa(wh0, ps1)
GP1 = transform_sample_counts(GP1, function(x) 1E6 * x/sum(x))
mb <- GP1
library(DESeq2)

#install.packages("C:/Users/Brianne/Downloads/locfit_1.5-9.tar.gz", repos = NULL, type="source")

# First, we will convert the sequence count data from our phyloseq object into the proper format for DESeq2
mb.dds <- phyloseq_to_deseq2(mb, ~ Location)

# Next we need to estimate the size factors for our sequences
gm_mean = function(x, na.rm=TRUE){
    exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans = apply(counts(mb.dds), 1, gm_mean)
mb.dds = estimateSizeFactors(mb.dds, geoMeans = geoMeans)

# Finally, we can run the core DESeq2 algorithm
mb.dds = DESeq(mb.dds, fitType="local")

# Create a new dataframe from the results of our DESeq2 run
mb.res = results(mb.dds)

# Reorder the sequences by their adjusted p-values
mb.res = mb.res[order(mb.res$padj, na.last=NA), ]

# Set your alpha for testing significance, and filter out non-significant results
alpha = 0.05
mb.sigtab = mb.res[(mb.res$padj < alpha), ]

mb.res = cbind(as(mb.res, "data.frame"), as(tax_table(mb)[rownames(mb.res), ], "matrix"))

# Add taxonomy information to each sequence
mb.sigtab = cbind(as(mb.sigtab, "data.frame"), as(tax_table(mb)[rownames(mb.sigtab), ], "matrix"))
head(mb.sigtab)

mb.sigtab$group<- ifelse(mb.sigtab$log2FoldChange < 0, 'buried','floating')
mb.sigtab$color <- ifelse(mb.sigtab$log2FoldChange < 0, '#882255','#88CCEE')
mb.ns <- mb.res %>% filter(padj > 0.05)

volcano.plot <- ggplot(mb.sigtab, aes(x = log2FoldChange, y = -log10(padj))) + geom_point(color = mb.sigtab$color, size = 4) + geom_hline(aes(yintercept =-log10(0.05)), linetype = "dashed") + theme_bw() + geom_point(data = mb.ns, mapping = aes(x = mb.ns$log2FoldChange, y = -log10(mb.ns$padj)), color = "gray", size = 4) + ggrepel::geom_label_repel(aes(label = rownames(mb.sigtab)), max.overlaps = 20, size = 5, label.padding = 0.1) + theme(text=element_text(size=20)) 

pdf("figures/Experiment7/NymphaeaNuphar_Figures_April2023/volcanoplot_16S.pdf")
volcano.plot
dev.off()

```



```{r}
bugbase <- fread("/Users/briannepalmer/Library/CloudStorage/OneDrive-Personal/LeafFossilization/Leaf Fossilization R/Experiment 7/data/Experiment7/BugBaseWaterLilies.csv")
```


```{r}
compare_means(Rel ~ Treatment, data = bugbase, 
              group.by = "Function")

bugbase.plot <- bugbase %>% filter(Function != "GramPositive", Function != "GramNegative") 

stat.test <- bugbase.plot %>%
  group_by(Function) %>%
  wilcox_test(Rel ~ Treatment) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test <- stat.test %>% add_xy_position(x = "Treatment")
```

```{r, echo=FALSE}
bug.plot <- ggboxplot(bugbase.plot, x = "Treatment", y = "Rel", 
  facet.by = "Function", scales = "free", fill = "Treatment", nrow =1) +  
  stat_pvalue_manual(
    stat.test,
    label = "{p}"
    ) + scale_fill_manual(values = two.pal) + theme_bw() + theme(legend.position = "none") + theme(text=element_text(size=20))

pdf("figures/Experiment7/NymphaeaNuphar_Figures_April2023/bugbase.pdf", height = 20, width = 20)
bug.plot
dev.off()

```

```{r,echo=FALSE}
gram <- ggplot(bugbase %>% filter(Function == c("GramNegative", "GramPositive")), aes(x = Treatment, y = Rel/3, fill= Function)) + geom_bar(stat = "identity", position = "stack") + theme_bw()+ labs(x= "Location", y = "Relative Abundance", title = "D.") + scale_fill_manual(values = two.pal)+ theme(text=element_text(size=20))
gram
```

Now, onto PiCRUST2 

for more details see the file called "picrust_waterlilies.R" -- this is where I parsed a lot of the data and stored it in the data folder 

```{r}
p2_PW <- "/Users/briannepalmer/Downloads/picrust2-2.5.1/picrust2_out_stratified/pathways_out/path_abun_unstrat.tsv.gz" # pathways
p2_KO <- "/Users/briannepalmer/Downloads/picrust2-2.5.1/picrust2_out_stratified/KO_metagenome_out/pred_metagenome_unstrat.tsv.gz" # KEGG
p2_EC <-"/Users/briannepalmer/Downloads/picrust2-2.5.1/picrust2_out_stratified/EC_metagenome_out/pred_metagenome_unstrat.tsv.gz"
sam_tab <- "/Users/briannepalmer/Library/CloudStorage/OneDrive-Personal/LeafFossilization/NymphaeaNuphar_18S/sample-metadata.csv" # metadata
```

```{r}
# imports PiCRUST data as phyloseq object
ps.pw <- import_picrust2(
  p2_PW,
  sam_tab,
  trait = "PATHWAY")

random_tree = rtree(ntaxa(ps.pw), rooted=TRUE, tip.label=taxa_names(ps.pw))
ps.pw = merge_phyloseq(ps.pw,random_tree)
ps.pw@sam_data <- sample_data(metadata)
rownames(ps@sam_data) <- ps@sam_data$SampleID
colnames(ps@otu_table) <- ps@sam_data$SampleID
```

```{r, echo=FALSE}
wh0 = genefilter_sample(ps.pw, filterfun_sample(function(x) x > 5), A=0.5*nsamples(ps)) # removes low abundance pathways
GP1 = prune_taxa(wh0, ps.pw)
GP1 = transform_sample_counts(GP1, function(x) 1E6 * x/sum(x))
mb <- GP1
library(DESeq2)

#install.packages("C:/Users/Brianne/Downloads/locfit_1.5-9.tar.gz", repos = NULL, type="source")

# First, we will convert the sequence count data from our phyloseq object into the proper format for DESeq2
mb.dds <- phyloseq_to_deseq2(mb, ~ Location)

# Next we need to estimate the size factors for our sequences
gm_mean = function(x, na.rm=TRUE){
    exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans = apply(counts(mb.dds), 1, gm_mean)
mb.dds = estimateSizeFactors(mb.dds, geoMeans = geoMeans)

# Finally, we can run the core DESeq2 algorithm
mb.dds = DESeq(mb.dds, fitType="local")


# Create a new dataframe from the results of our DESeq2 run
mb.res = results(mb.dds)

# Reorder the sequences by their adjusted p-values
mb.res = mb.res[order(mb.res$padj, na.last=NA), ]

# Set your alpha for testing significance, and filter out non-significant results
alpha = 0.05
mb.sigtab = mb.res[(mb.res$padj < alpha), ]

mb.res = cbind(as(mb.res, "data.frame"), as(tax_table(mb)[rownames(mb.res), ], "matrix"))

# Add taxonomy information to each sequence
mb.sigtab = cbind(as(mb.sigtab, "data.frame"), as(tax_table(mb)[rownames(mb.sigtab), ], "matrix"))
head(mb.sigtab)

mb.sigtab$group<- ifelse(mb.sigtab$log2FoldChange < 0, 'buried','floating')
mb.sigtab$color <- ifelse(mb.sigtab$log2FoldChange < 0, '#882255','#88CCEE')
mb.ns <- mb.res %>% filter(padj > 0.05)

volcano.plot <- ggplot(mb.sigtab, aes(x = log2FoldChange, y = -log10(padj))) + geom_point(color = mb.sigtab$color, size = 4) + geom_hline(aes(yintercept =-log10(0.05)), linetype = "dashed") + theme_bw() + ggtitle("B.") + geom_point(data = mb.ns, mapping = aes(x = mb.ns$log2FoldChange, y = -log10(mb.ns$padj)), color = "gray", size = 4) + ggrepel::geom_label_repel(aes(label = rownames(mb.sigtab)), max.overlaps = 10, size = 5, label.padding = 0.1) + theme(text=element_text(size=20)) 

volcano.plot

```

Ordnation
```{r, echo=FALSE}
ordu = ordinate(GP1, "PCoA", "unifrac", weighted=TRUE)
pw.ord <- plot_ordination(GP1, ordu, color="Location", shape="plant") + theme_bw() + labs(x = "PCOA1", y = "PCOA2")  + geom_point(size =5) + ggtitle("C.") + scale_color_manual(values = two.pal) + theme(text=element_text(size=20))
pw.ord
```

STATISTICS TIME! (PERMANOVA)
```{r}
set.seed(12345)
adonis2(phyloseq::distance(GP1, method="bray") ~ Location,
       data = metadata) # 0.352
adonis2(phyloseq::distance(GP1, method="bray") ~ plant,
        data = metadata) # 0.37
```
